pkg/opensearch/lucene_handler.go

```diff
@@ -279,14 +280,16 @@ func (h *luceneHandler) executeQueries(ctx context.Context) (*backend.QueryDataR
 		return nil, nil
 	}
 
+	response := backend.NewQueryDataResponse()
 	req, err := h.ms.Build()
 	if err != nil {
-		return nil, err
+		return errorsource.AddPluginErrorToResponse(h.queries[0].RefID, response, err), nil
 	}
 
 	res, err := h.client.ExecuteMultisearch(ctx, req)
 	if err != nil {
-		return nil, err
+		// We are returning the error containing the source that was added through errorsource.Middleware
+		return errorsource.AddErrorToResponse(h.queries[0].RefID, response, err), nil
```

Comment:
Should we maybe draw out the `h.queries[0].RefID` into a defined reusable var in this case?